File: in_Python/1189 Maximum Number of Balloons.py
class Solution(object):
    def maxNumberOfBalloons(self, text):
        """
        :type text: str
        :rtype: int
        """
        dict = {}
        for s in list(text):
            if s in ['b','a','l','o','n']:
                if s not in dict:
                    dict[s] = 1
                else:
                    dict[s] += 1
        t = []
        for s in ['b','a','n','o','l']:
            if s not in dict:
                return 0
            else:
                t.append(dict[s])
        a = min(t[0:3])
        b = min(t[3:])
        if b > a*2:
            return a
        else:
            return b/2

File: in_Java/1189 Maximum Number of Balloons.java
class Solution {
    public int maxNumberOfBalloons(String text) {
        Map<Character,Integer> hm = new HashMap<>();
        for(char c : text.toCharArray()){
            if(c == 'b' || c == 'a' || c == 'l' || c == 'o' || c == 'n'){
                if(!hm.containsKey(c)) hm.put(c,1);
                else hm.put(c, hm.get(c)+1);
            }
        }
        if(hm.keySet().size() < 5) return 0;
        int a = Math.min(hm.get('n'),Math.min(hm.get('b'), hm.get('a')));
        int b = Math.min(hm.get('l'), hm.get('o'));
        return Math.min(a, b/2);
    }
}